### PR TITLE
[Blueprint] Fix user identity in blueprint assignment

### LINKED ISSUE #2381: assignment with user-identity
## Describe the bug

**Command Name**
`az blueprint assignment update
Extension Name: blueprint. Version: 0.1.1.`

**Errors:**
```
Unable to build a model: Unable to deserialize response data. Data: ['/subscriptions/9f89d8cc-c662-4695-b270-57c92157bdb5/resourceGroups/hip-networking/providers/Microsoft.ManagedIdentity/userAssignedIdentities/azure-blueprints-cross-subscription'], {UserAssignedIdentity}, TypeError: string indices must be integers, DeserializationError: Unable to deserialize response data. Data: ['/subscriptions/9f89d8cc-c662-4695-b270-57c92157bdb5/resourceGroups/hip-networking/providers/Microsoft.ManagedIdentity/userAssignedIdentities/azure-blueprints-cross-subscription'], {UserAssignedIdentity}, TypeError: string indices must be integers
Traceback (most recent call last):
...
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az blueprint assignment update --name {} --location {} --blueprint-version {} --locks-mode {} --parameters {} --description {} --subscription {} --identity-type {} --user-assigned-identities {}`

## Environment Summary
```
Linux-5.4.0-47-generic-x86_64-with-debian-bullseye-sid
Python 3.6.5
Installer: DEB

azure-cli 2.3.1 *
azure-cli-ml 1.3.0

Extensions:
blueprint 0.1.1
synapse 0.3.0
azure-cli-ml 1.3.0

```

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Fix #2381, #2248

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/blueprint/azext_blueprint/_params.py [hunk 1]
```diff
@@ -219,10 +219,7 @@ def load_arguments(self, _):
             from .action import ResourceGroupAssignAddAction
             c.argument('assignment_name', options_list=['--name', '-n'], help='Name of the blueprint assignment.')
             c.argument('location', arg_type=get_location_type(self.cli_ctx))
-            c.argument('identity_type', arg_type=get_enum_type(ManagedServiceIdentityType), default='SystemAssigned', arg_group='Identity', help='Type of the managed identity.')
-            c.argument('identity_principal_id', options_list=['--principal-id'], arg_group='Identity', help='Azure Active Directory principal ID associated with this Identity.')
-            c.argument('identity_tenant_id', options_list=['--tenant-id'], arg_group='Identity', help='ID of the Azure Active Directory.')
```
Comment (fengzhou-msft) [inline src/blueprint/azext_blueprint/_params.py:224]: These 2 are only needed in response.

**Comment by yonzhan** [comment]
Blueprint

File: src/blueprint/azext_blueprint/_validators.py [hunk 2]
```diff
@@ -20,3 +20,8 @@ def blueprint_assignment_validator(cmd, namespace):
     if namespace.management_group:
         raise CLIError("The management group scope for blueprint assignment is not supported yet. Please use --subscription for subscription scope.")
     namespace.scope = '/subscriptions/{}'.format(namespace.subscription if namespace.subscription else get_subscription_id(cmd.cli_ctx))
+    try:
+        if namespace.user_assigned_identity and not namespace.identity_type:
+            namespace.identity_type = 'UserAssigned'
+    except AttributeError:
+        pass
```
Comment (qwordy) [inline src/blueprint/azext_blueprint/_validators.py:27]: Add a warning?

File: src/blueprint/azext_blueprint/tests/latest/test_blueprint_scenario.py [hunk 3]
```diff
@@ -105,6 +116,29 @@ def test_blueprint(self, resource_group):
...
 
+        with mock.patch('azure.cli.command_modules.role.custom._gen_guid', side_effect=self.create_guid):
+            self.cmd(
+                'az role assignment create '
+                '--assignee-object-id {identityPrincipalId} '
+                '--role Owner '
+                '--scope /subscriptions/{subscription}/resourceGroups/{rgName}')
+
+        import time
+        time.sleep(600)
```
Comment (qwordy) [inline src/blueprint/azext_blueprint/tests/latest/test_blueprint_scenario.py:127]: magic number

**Review by qwordy — APPROVED** [review]

File: src/blueprint/azext_blueprint/_validators.py (hunk 2 above)
Comment (fengzhou-msft) [reply src/blueprint/azext_blueprint/_validators.py:27]: The validator is in the scope of `blueprint assignment` while the attribute of `identity_type` only exists in `create/update`. So this error is expected for commands like `show`.

File: src/blueprint/azext_blueprint/tests/latest/test_blueprint_scenario.py (hunk 3 above)
Comment (fengzhou-msft) [reply src/blueprint/azext_blueprint/tests/latest/test_blueprint_scenario.py:127]: Yes, we may need better approach to wait for role assignment to take effective. I can't think of better ways at this moment.
https://docs.microsoft.com/en-us/azure/role-based-access-control/troubleshooting#role-assignment-changes-are-not-being-detected

Comment (bim-msft) [inline src/blueprint/azext_blueprint/tests/latest/test_blueprint_scenario.py:127]: Better to do it this way:
```suggestion
        if self.is_live:
            time.sleep(600)

```

Comment (fengzhou-msft) [reply src/blueprint/azext_blueprint/tests/latest/test_blueprint_scenario.py:127]: According to @qwordy, `time.sleep()` is automatically mocked in playback test.
https://github.com/Azure/azure-cli/blob/00cf5e3b3ecadfa43d4b1bee8c5c2d352626d227/src/azure-cli-testsdk/azure/cli/testsdk/base.py#L108

**Review by haroldrandom — APPROVED** [review]

File: src/blueprint/azext_blueprint/tests/latest/test_blueprint_scenario.py (hunk 3 above)
Comment (bim-msft) [reply src/blueprint/azext_blueprint/tests/latest/test_blueprint_scenario.py:127]: @fengzhou-msft Got it! It's great!

**Review by bim-msft — APPROVED** [review]

**Pull request merged**